modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/transfer/ProcessImport.java

```diff
@@ -78,72 +78,76 @@ public void importProcesses(InputStream processZipInputStream, String user)
-                        new File(dirPath).mkdirs();
-                        FileOutputStream fileOutputStream = null;
-                        try {
-                            fileOutputStream = new FileOutputStream(outpath);
-                            int len = 0;
-                            while ((len = zipInputStream.read(buffer)) > 0) {
-                                fileOutputStream.write(buffer, 0, len);
-                            }
-                        } finally {
-                            if (fileOutputStream != null) {
-                                fileOutputStream.close();
+            try {
+                if (importsDir != null) {
+                    importsDir.mkdirs();
+
+                    try {
+                        ZipEntry entry;
+                        while ((entry = zipInputStream.getNextEntry()) != null) {
+                            //counter++;
+                            String outpath = ProcessCenterConstants.IMPORTS_DIR + "/" + entry.getName();
+                            String dirPath = outpath.substring(0, outpath.lastIndexOf("/"));
+                            new File(dirPath).mkdirs();
+                            FileOutputStream fileOutputStream = null;
+                            try {
+                                fileOutputStream = new FileOutputStream(outpath);
+                                int len = 0;
+                                while ((len = zipInputStream.read(buffer)) > 0) {
+                                    fileOutputStream.write(buffer, 0, len);
+                                }
+                            } finally {
+                                if (fileOutputStream != null) {
+                                    fileOutputStream.close();
+                                }
                             }
                         }
+                    } finally {
+                        zipInputStream.close();
                     }
-                } finally {
-                    zipInputStream.close();
-                }
 
-                //check if the importing processes are already available
-                boolean isProcessesAvailableAlready = isProcessesAlreadyAvailble();
-                if (isProcessesAvailableAlready) {
-                    log.error(
-                            "One or more process in the importing archive is already cavailable in the Process Center");
-                    throw new ProcessCenterException("ALREADY_AVAILABLE");
-                }
+                    //check if the importing processes are already available
+                    boolean isProcessesAvailableAlready = isProcessesAlreadyAvailble();
+                    if (isProcessesAvailableAlready) {
+                        log.error(
+                                "One or more process in the importing archive is already cavailable in the Process Center");
+                        throw new ProcessCenterException("ALREADY_AVAILABLE");
```

Comment:
There is a typo in the error message above. Also why the error message we log and throw are different